src/lib.rs

```diff
@@ -0,0 +1,54 @@
+#[macro_use]
+extern crate nickel;
+extern crate openssl;
+extern crate rand;
+extern crate rustc_serialize;
+extern crate byteorder;
+extern crate time;
+extern crate cookie;
+extern crate plugin;
+extern crate typemap;
+
+use std::fmt::Debug;
+use rustc_serialize::{Encodable, Decodable};
+use std::any::Any;
+use nickel::{Response, Middleware, MiddlewareResult, cookies, NickelError, Session, SessionStore};
+use nickel::status::StatusCode::Forbidden;
+
+pub type ValidateUserFunc<T> = Box<Fn(&T) -> bool + Send + Sync>;
+
+pub struct Authorizer<T, D> {
+    authorize: ValidateUserFunc<T>,
+    access_granted: Box<Middleware<D>+ Sized + Send + Sync>,
+}
+
+impl<T, D> Authorizer<T, D> {
+    pub fn new(authorize: ValidateUserFunc<T>,
+               access_granted: Box<Middleware<D> + Sized + Send + Sync>)
+               -> Authorizer<T, D> {
+        Authorizer { authorize: authorize, access_granted: access_granted }
+    }
+}
+
+impl< T, D: 'static> Middleware<D> for Authorizer<T, D>
+    where
+    D: AsRef<cookies::SecretKey> + SessionStore<Store=T>,
+    T: 'static + Any + Encodable + Decodable + Default + Debug
```

Comment:
Regarding `current_user`, I suppose that can be nice.

Could something like:
```
trait SessionUser{
    fn current_user(&self) -> Option<String>;
}
```
work? The user would implement this for whatever session data they store and we can provide some default implementation for `String` for example.